sklearn/mixture/_gaussian_mixture.py

```diff
@@ -715,10 +715,10 @@ def _initialize(self, X, resp):
         weights, means, covariances = _estimate_gaussian_parameters(
             X, resp, self.reg_covar, self.covariance_type
         )
-        weights /= n_samples
 
         self.weights_ = weights if self.weights_init is None else self.weights_init
         self.means_ = means if self.means_init is None else self.means_init
+        self.weights_ /= self.weights_.sum()
```

Comment:
Can `weights` be negative? If so, I think this might be more appropriate:
```suggestion
        self.weights_ /= np.abs(self.weights_).sum()
```

cc @jeremiedbb which whom we discussed normalization with negative weights.

Reply:
IIRC `self.weights_` should be non-negative at this point. Not completely sure. I will try to take a look later, time permitting (though this isn't too high on my list)

Reply:
Side comment: does anyone know of applications that are wide spread and use negative sample weights?

The only people I know who have negative sample weights are (some) researchers in High Energy Particle physics. From my memory the interpretation of what those samples with negative weights mean is tricky. The consensus I remember is that a group of samples should always have a net positive weight (but this isn't always true), but that you shouldn't take the absolute value of a individual sample's weight. The reason for not using `abs(weight)` is that a possible interpretation of a negative sample weight is that you are trying to "subtract" that sample from your final group of samples. (I think the "deep physics" reasons for how you end up with them is that quantum mechanics has constructive and destructive interference terms, and depending on how your sample generator is written you end up with samples with a negative weight that are meant to represent negative interference, or something like that.)

Basically, are negative weights something worth spending time on or best to punt on it till there are use-cases?

Reply:
> Basically, are negative weights something worth spending time on or best to punt on it till there are use-cases?

AFAIK not really, but I wonder what prompted @jjerphan and @jeremiedbb's conversations on normalizing negative weights.

Reply:
What @betatim described prompted those discussions; I think @betatim definitely is more knowledgeable than me on this subject.

I think, use-case relying on negative weights are relatively rare, and this might not be something we would like to maintain relevant normalization for if this normalization is complex.

Reply:
I think it is good to assume that weights are always positive and use that to reduce complexity/additional computation.

Reply:
As long as we avoid (re-normalizing) user provided `weights_init` (see my review below), with the current capabilities of Mixture classes like GMM, I don't see how its possible to get negative weights. Thus I hope this isn't something we have to consider one way or another. 

And I would really push to not re-normalize user provided `weights_init`, and especially to not make rigid decisions on how to re-normalize negative vs. positive weights, etc. Otherwise we take away the ability to precisely define an initial state for the EM algorithm.

Reply:
The negative weights come back all the time ;). The last report was the following: https://github.com/scikit-learn/scikit-learn/issues/18934

If really only want positive weight, we should use `only_non_negative=True` in `_check_sample_weight`. I assume that we should come up with a PR allowing negative weight for people that really want it but without any guarantee that the normalization makes any sense. If an algorithm was written to take into account negative weight then we can always make sure that the normalization is properly done as per the literature.